language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ClassForType.scala

```diff
@@ -46,11 +46,22 @@ object ClassForType extends StrictLogging {
     for {
       (interfaceName, interface) <- typeWithContext.interface.interfaces.toList
       className = ClassName.bestGuess(fullyQualifiedName(interfaceName))
+      interfaceViewTypeName = ClassName.bestGuess(
+        fullyQualifiedName(
+          interface.viewType.getOrElse(
+            throw new IllegalArgumentException(
+              s"View Type is required for interface $interfaceName"
+            )
```

Comment:
Is it correct to assume view type is required for all interfaces?

Reply:
It is correct. Moreover, it is always a type `T` that `extends codegen.DamlRecord<T>`, if you have a reason to exploit that.